tsdb/engine/tsm1/encoding.gen.go.tmpl

```diff
@@ -53,18 +53,23 @@ func (a {{.Name}}Values) assertOrdered() {
 // Deduplicate returns a new slice with any values that have the same timestamp removed.
 // The Value that appears last in the slice is the one that is kept.
 func (a {{.Name}}Values) Deduplicate() {{.Name}}Values {
-	m := make(map[int64]{{.Name}}Value)
-	for _, val := range a {
-		m[val.UnixNano()] = val
+	if len(a) == 0 {
+		return a
 	}
+	sort.Stable(a)
```

Comment:
This could be more expensive than an unstable sort. Do we need it to be stable given we're only going to be removing duplicate elements anyway?